sdk/azidentity/example_user_auth_test.go

```diff
@@ -39,20 +37,27 @@ func storeRecord(record azidentity.AuthenticationRecord) error {
 // interactively every time the application runs. The example uses [InteractiveBrowserCredential], however
 // [DeviceCodeCredential] has the same API. The key steps are:
 //
-//  1. Enable persistent caching by importing "github.com/Azure/azure-sdk-for-go/sdk/azidentity/cache" and
-//     setting [TokenCachePersistenceOptions]
-//  2. Call Authenticate to acquire an [AuthenticationRecord] and store that for future use. An [AuthenticationRecord]
+//  1. Call [github.com/Azure/azure-sdk-for-go/sdk/azidentity/cache.New] to construct a persistent cache
+//  2. Set the Cache field in the credential's options
+//  3. Call Authenticate to acquire an [AuthenticationRecord] and store that for future use. An [AuthenticationRecord]
 //     enables credentials to access data in the persistent cache. The record contains no authentication secrets.
-//  3. Add the [AuthenticationRecord] to the credential's options
+//  4. Add the [AuthenticationRecord] to the credential's options
 func Example_persistentUserAuthentication() {
 	record, err := retrieveRecord()
 	if err != nil {
 		// TODO: handle error
 	}
+	c, err := cache.New(nil)
+	if err != nil {
+		// TODO: handle error. An error here means persistent
```

Comment:
Why don't we simply error out. Continuing on seems bad.

Reply:
Whether to continue is up to developers. Continuing means the application can do what the user asked but will require the user to log in again the next time it runs. I show that in the example because I believe most developers would prefer it to the alternative, which is the app being unable to access Azure resources.